### ISSUE TITLE
use multiple `model_serializer`s

### ISSUE THREAD

**Issue by ornariece** [pydantic/pydantic#9063]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Using multiple `@model_serializer`s in the same model results in every ``model_serializer`` but the last-declared one being discarded.
Is there any reason behind this behavior, beyond the difficulty of implementing the correct handling of multiple ``model_serializer``s? While I can imagine the implementation would not be trivial, I don't see how the concept doesn't make sense.
If you explicitly want to prohibit having multiple ``model_serializer``s, it should probably be enforced just like with ``field_serializer``, otherwise it's up to the user to figure out that only the last ``model_serializer`` is being used. But I don't think it should be prohibited.

### Example Code

```Python
import pydantic


class MyModel(pydantic.BaseModel):
    x: int

    @pydantic.model_serializer(mode='wrap')  # this serializer is never used
    def serializer_A(
        self, handler: pydantic.SerializerFunctionWrapHandler
    ) -> dict[str, int]:
        dumped = handler(self)
        dumped['y'] = dumped['x']
        return dumped

    @pydantic.model_serializer(mode='wrap')
    def serializer_B(
        self, handler: pydantic.SerializerFunctionWrapHandler
    ) -> dict[str, dict[str, int]]:
        return {'wrapped': handler(self)}


if __name__ == '__main__':
    my_model = MyModel(x=1)

    print(my_model.model_dump())
    # result: {'wrapped': {'x': 1}}
    # expected: {'wrapped': {'x': 1, 'y': 1}}
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.7.0a1                                                                                          pydantic-core version: 2.16.2                                                                                             pydantic-core build: profile=release pgo=false                                                                                 install path: C:\Users\me\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydantic                          python version: 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]                              platform: Windows-10-10.0.19045-SP0                                                                             related packages: email-validator-2.0.0.post2 mypy-1.8.0 pyright-1.1.352 typing_extensions-4.10.0                                 commit: unknown
```

**Comment by sydney-runkle** [comment]
@ornariece,

Thanks for your question! I'm currently leaning towards raising a `PydanticUserError` if multiple `model_serializer`s are defined, like you mentioned we do for `field_serializer`s.

Regarding the reasoning behind why we couldn't support multiple model serializers, I'm most concerned about the abundance of room for user error. AFAIK, it doesn't make sense to define multiple `plain` model serializers, though I understand the case for multiple layers in a `wrap` serializer. 

It also seems as though if you really want to perform multiple wrapping operations with serializers, you could just do all of that in one `wrap` serializer with multiple steps / iterations.

Looking forward to hearing your thoughts!

**Comment by sydney-runkle** [comment]
Closing for now! Ping me if you think we should leave this open given the above considerations.

**Comment by ornariece** [comment]
@sydney-runkle,

i get the concern about room for user error. i'd argue that if the user were to mistakenly declare multiple `model_serializer`s then that's on them?
in my usecase, it seemed natural to rely on the fact that one serializer was declared after the other, so that the serializers would be applied sequentially. in fact, this is exactly what happens when you declare mutliple `'wrap'` `model_validator`s! and there too you can make the exact same argument about room for user error.

i can show you my usecase in which having multiple `model_serializer`s felt natural to me, so much so that i never paused to think about if pydantic supports it. that said, the workaround around the current behavior for this usecase is very bearable, because as you said "you could just do all of that in one wrap serializer with multiple steps / iterations".

in conclusion, i'd still argue that declaring multiple `model_serializer`s should be a thing, just like declaring multiple `model_validator`s is a thing, but it might not be of utmost importance right now given that the workaround is trivial while the implementation of the change probably isn't.
